cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -982,6 +1021,407 @@ template <typename OutType, typename ArgType, typename Op>
 using ScalarBinaryNotNullStatefulEqualTypes =
     ScalarBinaryNotNullStateful<OutType, ArgType, ArgType, Op>;
 
+// A kernel exec generator for ternary functions that addresses both array and
+// scalar inputs and dispatches input iteration and output writing to other
+// templates
+//
+// This template executes the operator even on the data behind null values,
+// therefore it is generally only suitable for operators that are safe to apply
+// even on the null slot values.
+//
+// The "Op" functor should have the form
+//
+// struct Op {
+//   template <typename OutValue, typename Arg0Value, typename Arg1Value, typename
+//   Arg2Value> static OutValue Call(KernelContext* ctx, Arg0Value arg0, Arg1Value arg1,
+//   Arg2Value arg2, Status *st) {
```

Comment:
Could you format this?
This is difficult to read.

Reply:
Reformatted.